_posts/henry_zhao/2019-03-06-Implement-Stardust.md

```diff
@@ -0,0 +1,287 @@
+---
+layout: post
+title: 'Accelerate data visualization with WebGL Platform Stardust'
+breadcrumb: true
+author: henry_zhao
+date: 2019-03-06
+categories: henry_zhao
+tags:
+  - Javascript
+  - WebGL
+  - Stardust
+teaser:
+  info: Data visualization with massive data could be fine in your browser
+  image: henry_zhao/stardust.png
+header:
+  version: small
+  title: Software Engineering Blog
+  image: header-logo-crop.png
+  icon: icon-blog
+---
+
+<image src="{{ site.urlimg }}/henry_zhao/future_world.jpg" />
+
+# Introduction
+
+Genomic data is huge and messy. In order to provide researchers a powerful data analysis&management platform, what we present data is the most popular data processing library, d3. It has a wealth of tools that can use svg to portray and manipulate data patterns to the greatest extent possible with different ranges and different forms of pie chart, histograms charts and human figures. For example:
+
+<image src="{{ site.urlimg }}/henry_zhao/human_body.png" />
+
+However, when the data set is too big(larger than 10000 node), there will an inevitable DOM operation bottleneck. It makes me think about what kind of tools can improve or even replace d3 to be able to visualize data faster.
+So first we should know what is svg and why it has bottleneck.
+
+### SVG
+
+SVG is a vector graphic format—based on XML and is used to display a variety of graphics on the Web and other environments. Since SVG are defined in XML, every SVG element is appended to the DOM which is big cost(It is the reason that reactjs use virtual DOM to avoid DOM operation).
+
+### WebGL
+
+There is a potential way to replace svg which is WebGl. WebGL (Web Graphics Library) is a JavaScript API for rendering interactive 3D and 2D graphics within any compatible web browser without the use of plug-ins. WebGL does so by introducing an API that closely conforms to OpenGL ES2.0 that can be used in HTML5 canvas elements. It provides local computing resources API to accelerate the rendering of the graphics and avoids massive operations of DOM. Therefore, it is naturally suitable for large data manipulation.
+
+[WebGL Example](https://webglsamples.org/)
+
+<image src="{{ site.urlimg }}/henry_zhao/webgl_building.gif" />
+
+There are a lot of WebGL libraries for data visualization. I will use stardust[stardust](https://stardustjs.github.io/) as an example to demonstrate how to implement it.
+
+### Stardust
+
+First install stardust:
```

Comment:
Are you missing npm install ? The command below is importing stardust, not installing.